### PR TITLE
Use mathematical function to determine yaw direction

### PR DESCRIPTION
**Pull request by sfan5**
Fixed version of #4235 by @Ambistic

(the branch being in the wrong place is GitHubs fault)

### CODE REVIEW

**Comment by sofar** [comment]
Looks OK to me.

**Comment by paramat** [comment]
:+1:

File: src/game.cpp [hunk 1]
```diff
@@ -4296,23 +4296,13 @@ void Game::updateFrame(ProfilerGraph *graph, RunStats *stats,
 
 inline static const char *yawToDirectionString(int yaw)
 {
-	// NOTE: TODO: This can be done mathematically without the else/else-if
-	// cascade.
-
 	const char *player_direction;
+	std::string direction[4] = {"North [+Z]", "West [-X]", "South [-Z]", "East [+X]"}
```
Comment (est31) [inline src/game.cpp]: Ermm where is the semicolon? This doesn't compile

**Comment by est31** [comment]
@sfan5 you should fix all the compiler warnings and errors before pushing this.

File: src/game.cpp (hunk 1 above)
Comment (paramat) [inline src/game.cpp]: Oops missed that error.
Naughty not testing this.

File: src/game.cpp [hunk 2]
```diff
@@ -4296,23 +4296,13 @@ void Game::updateFrame(ProfilerGraph *graph, RunStats *stats,
...
-		player_direction = "West [-X]";
-	else if (yaw >= 135 && yaw < 225)
-		player_direction = "South [-Z]";
-	else if (yaw >= 225 && yaw < 315)
-		player_direction = "East [+X]";
-	else
-		player_direction = "North [+Z]";
-
-	return player_direction;
+	return direction[yaw];
```
Comment (Zeno-) [inline src/game.cpp]: What are the conversion and scope rules regarding returning a local variable of type std::string as a const char*? 

I'd rather see the array simply as `static const char *direction[4] = { "North [+Z]", "West [-X]", "South [-Z]", "East [+X]" };`

File: src/game.cpp [hunk 3]
```diff
@@ -4297,7 +4297,7 @@ void Game::updateFrame(ProfilerGraph *graph, RunStats *stats,
 inline static const char *yawToDirectionString(int yaw)
 {
 	const char *player_direction;
-	std::string direction[4] = {"North [+Z]", "West [-X]", "South [-Z]", "East [+X]"}
+	const char *direction[4] = {"North [+Z]", "West [-X]", "South [-Z]", "East [+X]"};
```
Comment (Zeno-) [inline src/game.cpp]: It needs to be `static const char *direction[4] = {"North [+Z]", "West [-X]", "South [-Z]", "East [+X]"};`

**Comment by Zeno-** [comment]
+1

**Comment by nerzhul** [comment]
:+1:

**Comment by est31** [comment]
7a532056e29d00006382caad9ccde66e54ca8aa5

**Pull request closed without merging**